File: Chapter 06/JS/section_6_4_3_the_modern_architect_ai_rule/domain/product.js
/**
 * THE DOMAIN MODEL (Entity).
 * Represents the internal business reality (e.g., a Database Table).
 */
class Product {
    constructor(id, name, isDigital, weightInLbs, price) {
        this.id = id;
        this.name = name;
        this.isDigital = isDigital;
        this.weightInLbs = weightInLbs;
        this.price = price;
    }
}

module.exports = Product;

File: Chapter 06/C#/Section_6_4_3_The_Modern_Architect_AI_Rule/Domain/Product.cs
namespace Chapter06.AiApiExample.Domain
{
    /// <summary>
    /// THE DOMAIN MODEL (Entity).
    /// Represents the internal business reality (e.g., a Database Table).
    /// This should NEVER be sent directly over the HTTP API.
    /// </summary>
    public class Product
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public bool IsDigital { get; set; }
        public decimal WeightInLbs { get; set; }
        public decimal Price { get; set; }
    }
}